examples/name-service/programs/name-service/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+        let signer_seeds: [&[u8]; 2] = [signer_seed, &[bump]];
+
+        let inputs = InstructionDataInvokeCpi {
+            proof: Some(proof),
+            new_address_params: vec![],
+            input_compressed_accounts_with_merkle_context: vec![compressed_account],
+            output_compressed_accounts: vec![new_compressed_account],
+            relay_fee: None,
+            compression_lamports: None,
+            is_compress: false,
+            signer_seeds: signer_seeds.iter().map(|seed| seed.to_vec()).collect(),
+            cpi_context,
+        };
+        let inputs = inputs.try_to_vec()?;
+
+        let signer_seeds: [&[&[u8]]; 1] = [&signer_seeds[..]];
+        let mut cpi_ctx = CpiContext::new_with_signer(
+            ctx.accounts.light_system_program.to_account_info(),
+            cpi_accounts(&ctx),
+            &signer_seeds[..],
+        );
+
+        cpi_ctx.remaining_accounts = ctx.remaining_accounts.to_vec();
+
+        light_system_program::cpi::invoke_cpi(cpi_ctx, inputs)?;
+
+        Ok(())
+    }
+
+    pub fn delete_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, NameService<'info>>,
+        compressed_account: PackedCompressedAccountWithMerkleContext,
+        proof: CompressedProof,
+        cpi_context: Option<CompressedCpiContext>,
+    ) -> Result<()> {
+        let signer_seed = b"cpi_signer".as_slice();
+        let bump = Pubkey::find_program_address(&[signer_seed], &ctx.accounts.self_program.key()).1;
+        let signer_seeds: [&[u8]; 2] = [signer_seed, &[bump]];
+
+        let inputs = InstructionDataInvokeCpi {
+            proof: Some(proof),
+            new_address_params: vec![],
+            input_compressed_accounts_with_merkle_context: vec![compressed_account],
+            output_compressed_accounts: vec![],
+            relay_fee: None,
+            compression_lamports: None,
+            is_compress: false,
+            signer_seeds: signer_seeds.iter().map(|seed| seed.to_vec()).collect(),
+            cpi_context,
+        };
+        let inputs = inputs.try_to_vec()?;
+
+        let signer_seeds: [&[&[u8]]; 1] = [&signer_seeds[..]];
+        let mut cpi_ctx = CpiContext::new_with_signer(
+            ctx.accounts.light_system_program.to_account_info(),
+            cpi_accounts(&ctx),
+            &signer_seeds[..],
+        );
+
+        cpi_ctx.remaining_accounts = ctx.remaining_accounts.to_vec();
+
+        light_system_program::cpi::invoke_cpi(cpi_ctx, inputs)?;
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug, AnchorDeserialize, AnchorSerialize)]
+pub enum RData {
+    A(Ipv4Addr),
+    AAAA(Ipv6Addr),
+    CName(String),
+}
+
+#[derive(Debug)]
+#[account]
+pub struct NameRecord {
+    pub owner: Pubkey,
+    pub name: String,
+    pub rdata: RData,
+}
+
+#[error_code]
+pub enum CustomError {
+    #[msg("No authority to perform this action")]
+    Unauthorized,
+}
+
+#[derive(Accounts)]
```

Comment:
[example use](https://github.com/Lightprotocol/light-protocol/blob/main/examples/token-escrow/programs/token-escrow/src/escrow_with_compressed_pda/escrow.rs#L29-L47) of the `light_accounts` and `light_traits` macros. so you only have to define custom accounts. 

same here -- please let me know whether the macros inline docs are straightforward, and what could be improved on these

Reply:
I submitted a small fix for `light_accounts` - https://github.com/Lightprotocol/light-protocol/pull/1063

About `light_traits` - it works, but I will give it a thought whether it can be improved.